vespalib/src/vespa/vespalib/btree/btreeiterator.hpp

```diff
@@ -656,28 +651,28 @@ lower_bound(BTreeNode::Ref rootRef, const KeyType & key, CompareT comp)
 
 template <typename KeyT, typename DataT, typename AggrT, typename CompareT,
           typename TraitsT>
+template <bool STRICT_SEEK>
 void
 BTreeConstIterator<KeyT, DataT, AggrT, CompareT, TraitsT>::
 seek(const KeyType & key, CompareT comp)
 {
     if (TraitsT::BINARY_SEEK) {
-        binarySeek(key, comp);
+        binarySeek<STRICT_SEEK>(key, comp);
     } else {
-        linearSeek(key, comp);
+        linearSeek<STRICT_SEEK>(key, comp);
     }
 }
 
 template <typename KeyT, typename DataT, typename AggrT, typename CompareT,
           typename TraitsT>
+template <bool STRICT_SEEK>
 void
 BTreeConstIterator<KeyT, DataT, AggrT, CompareT, TraitsT>::
 binarySeek(const KeyType & key, CompareT comp)
 {
     const LeafNodeType *lnode = _leaf.getNode();
     uint32_t lidx = _leaf.getIdx();
-#ifdef STRICT_BTREE_ITERATOR_SEEK
-    assert(_leaf.valid() && comp(lnode->getKey(lidx), key));
-#endif
+    assert( ! STRICT_SEEK || (_leaf.valid() && !comp(key, lnode->getKey(lidx))));
```

Comment:
Might as well remove the assert, instead of adding a template argument to the method.
Or add "assert_before" and "assert_not_after" methods on the iterator that a conservative caller can call.
The suggested change also changed the assert from "before" to "not after" check.